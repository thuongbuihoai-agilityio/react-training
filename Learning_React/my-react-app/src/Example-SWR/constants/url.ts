const BASE_URL = "http://localhost:1234";
const TASK_URL = `${BASE_URL}/task`;

export { BASE_URL, TASK_URL };
