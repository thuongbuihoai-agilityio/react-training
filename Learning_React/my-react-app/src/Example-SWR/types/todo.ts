export interface TodoType {
  id: string;
  title: string;
}

export interface TodoProps {
  todo: TodoType;
  deleteTodo: (id: string) => void;
}

export interface ModalUpdateProps {
  todo: TodoType;
  hideModalUpdate: Function;
  isReload: Boolean;
  setIsReLoad: Function;
}