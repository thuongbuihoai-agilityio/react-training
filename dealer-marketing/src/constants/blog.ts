import { Blog } from "@common-types/blog";

const BLOG_MOCKING: Blog = {
  blogId: "1",
  title: "Exploring Influential and Impactful Automotive Advertising Campaigns",
  slug: "exploring-influential-and-impactful-automotive-advertising-campaigns",
  createDate: "July 28, 2022",
  expertId: "Donna Welker",
  image: {
    url: "/images/past-present-future.png",
    alt: "This is image past present future",
  },
  description:
    "Advertising has come a long way since its humble beginnings. There's no doubt about it â€“ marketing remains an ever-changing landscape: What worked a few years ago may not be effective today, and what's popular now may be out of style in a few months. Nowhere is this more relevant than in the automotive industry, where new campaigns hold the potential to make or break a company. This article will explore some disruptive and impactful automotive marketing campaigns throughout the past century. We'll explore what made them successful and see how they changed the marketing landscape forever.",
};

export { BLOG_MOCKING };
