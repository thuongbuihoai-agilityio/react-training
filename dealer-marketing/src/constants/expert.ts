import { Expert } from "@common-types/expert";

const EXPERT_MOCKING: Expert = {
  expertId: "1",
  name: "Matt-Childers",
  slug: "matt-childers",
  info: "DealersLink",
  image: {
    url: "/images/avatar/matt-childers.png",
    alt: "This is image of Mr Matt-Childers",
  },
  description:
    "Brent Albrecht is currently the Vice President of Business Development at Friendemic. Friendemic provides social media and online reputation services for clients and agencies across the globe. Specialties: Marketing Strategy, Product development and launch, Social Media Marketing, Business Development, Lead Generation, Sales Promotion, Digital Marketing, Database Marketing.",
};

export { EXPERT_MOCKING };
