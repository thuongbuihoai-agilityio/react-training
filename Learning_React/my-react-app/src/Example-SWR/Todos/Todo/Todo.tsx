import { useCallback, useState } from "react";
import { Link } from "react-router-dom";
import Button from "../../components/Button/Button";
import { TodoProps } from "../../types/todo";
import ModalDelete from "../Modal/ModalDelete";

const Todo: React.FC<TodoProps> = ({ todo }) => {
  const [openModalDelete, setOpenModalDelete] = useState(false);
  const handleOpenModalDelete = useCallback(() => {
    setOpenModalDelete(true)
  }, []);

  return (
    <div data-testid="todo-item" className="todo__item">
      <ul className="task__list">
        <li className="task__item">
          <Link className="task__link" to={`/todo/${todo.id}`} state={{ todo }}>{todo.title}</Link>
          <Button value="Delete" onClick={handleOpenModalDelete} data-testid="delete-btn" />
        </li>
      </ul>
      {openModalDelete && <ModalDelete hideModalDelete={setOpenModalDelete} id={todo.id} />}
    </div>
  );
};

export default Todo;
