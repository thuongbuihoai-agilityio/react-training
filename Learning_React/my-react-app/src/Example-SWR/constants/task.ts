import { TodoType } from "../types/todo";

const TASK_MOCKING: TodoType = {
  id: "1",
  title: "Example create Todo App (20220522)"
};

export {TASK_MOCKING};