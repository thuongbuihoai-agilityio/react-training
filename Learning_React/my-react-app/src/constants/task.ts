import { TodoType } from "../Example-SWR/types/todo";

const TASK_MOCKING: TodoType = {
  id: "1",
  title: "Example create Todo App (20220522)"
};

export {TASK_MOCKING};