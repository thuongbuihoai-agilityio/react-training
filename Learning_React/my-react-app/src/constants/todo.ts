import { TodoProps } from "../Example-SWR/types/todo";

const TODO_CONSTANTS: TodoProps = {
  todo: {
    id: "a56403e9-8163-451c-a17b-6bd2fa3f389a",
    title: "Example create Todo App (20220522)",
  },
  deleteTodo: () => {}
};

export {
  TODO_CONSTANTS
}