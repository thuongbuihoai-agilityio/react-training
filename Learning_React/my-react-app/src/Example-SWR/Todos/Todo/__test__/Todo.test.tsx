import { act, fireEvent, render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import "@testing-library/jest-dom"
import { TODO_CONSTANTS } from "../../../../constants/todo";
import Todo from "../Todo";

describe("Todo item", () => {
  test("render Todo Item component", async () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Todo {...TODO_CONSTANTS} />
      </Router>
    );
    const todoItemElement = getByTestId("todo-item");
    expect(todoItemElement).toBeInTheDocument();
  });

  test("Should render pop when clicking delete button", async () => {
    const deleteTodo = jest.fn();
    const testingClassName = `pop-confirm-${Math.random}`;
    const { getByTestId } = render(
      <Todo
        {...TODO_CONSTANTS}
        deleteTodo={deleteTodo}
      />
    );

    const deleteBtn = getByTestId("delete-btn");
    act(() => {
      fireEvent.click(deleteBtn);
    });
    const popConfirmElm = document.getElementsByClassName(testingClassName)[0];
    expect(popConfirmElm).toBeInTheDocument();
  });
});