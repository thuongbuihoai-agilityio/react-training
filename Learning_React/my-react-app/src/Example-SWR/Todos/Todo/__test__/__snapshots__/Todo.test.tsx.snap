// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo item matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todo__item"
    data-testid="todo-item"
  >
    <ul
      class="task__list"
    >
      <li
        class="task__item"
      >
        <a
          class="task__link"
          href="/todo/a56403e9-8163-451c-a17b-6bd2fa3f389a"
        >
          Example create Todo App (20220522)
        </a>
        <button
          class="btn btn__del"
        >
          Delete
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
